sw/device/silicon_creator/lib/drivers/rstmgr.c

```diff
@@ -30,10 +32,39 @@ static void rstmgr_alert_info_collect(void) {
 }
 
 uint32_t rstmgr_reason_get(void) {
+  // Static assertions for bitfield indices.
+#define REASON_ASSERT(index, expect) \
+  static_assert((index) == (expect), #index " value incorrect.");
+  REASON_ASSERT(kRstmgrReasonPowerOn, RSTMGR_RESET_INFO_POR_BIT);
+  REASON_ASSERT(kRstmgrReasonLowPowerExit,
+                RSTMGR_RESET_INFO_LOW_POWER_EXIT_BIT);
+  REASON_ASSERT(kRstmgrReasonNonDebugModule, RSTMGR_RESET_INFO_NDM_RESET_BIT);
+  REASON_ASSERT(kRstmgrReasonSysrstCtrl,
+                RSTMGR_RESET_INFO_HW_REQ_OFFSET +
+                    kTopEarlgreyPowerManagerResetRequestsSysrstCtrlAonGscRst);
+  REASON_ASSERT(
+      kRstmgrReasonWatchdog,
+      RSTMGR_RESET_INFO_HW_REQ_OFFSET +
+          kTopEarlgreyPowerManagerResetRequestsAonTimerAonAonTimerRstReq);
+  // Alert escalation is one bit after the reset request index for the last
+  // peripheral.
+  REASON_ASSERT(kRstmgrReasonEscalation,
+                RSTMGR_RESET_INFO_HW_REQ_OFFSET +
+                    kTopEarlgreyPowerManagerResetRequestsLast + 1)
+  // Check that the last index corresponds to the last bit in HW_REQ.
+  REASON_ASSERT(
+      (1 << (kRstmgrReasonLast - RSTMGR_RESET_INFO_HW_REQ_OFFSET + 1)) - 1,
```

Comment:
the error message would look weird, you may want to use `static_assert` for this one.

Reply:
Yeah, you're right. Changed to use `static_assert` directly.